python-package/lightgbm/basic.py

```diff
@@ -3906,8 +3951,9 @@ def model_from_string(self, model_str: str) -> "Booster":
         self : Booster
             Loaded Booster object.
         """
-        if self._handle is not None:
-            _safe_call(_LIB.LGBM_BoosterFree(self._handle))
+        # ensure that existing Booster is freed before replacing it
+        # with a new one createdfrom file
+        _safe_call(_LIB.LGBM_BoosterFree(self._handle))
```

Comment:
As of #5947, there is no path through `lightgbm`'s public API which would allow a user to overwrite `Booster._handle` with `None`.

Making these changes to ensure it's never `None` avoids the need to do `if {booster}._handle is not None` types of checks in `_InnerPredictor.from_booster()`.